src/init.rs

```diff
@@ -59,3 +52,97 @@ pub fn init(shell_name: &str) {
         print!("{}", script);
     }
 }
+
+/* For bash: we need to manually hook functions ourself: PROMPT_COMMAND will exec
+   right before the prompt is drawn, and any function trapped by DEBUG will exec
+   before a command is run. There is no way to avoid clobbering these functions,
+   so we will not hook them if they are already defined.
+
+   There is a preexec/precmd framework for bash out there: if we find the
```

Comment:
That's correct. Actually, even if `bash-preexec` is installed on its own, it won't do anything because apparently you're supposed to source it as the very last thing in your shell config. Presumably if you were *using* the functionality, you'd have those arrays defined.

Honestly I don't know what the best way to do this is. I really want to avoid clobbering any pre-existing functionality, but there's no way to do that in native bash while also installing our timing functions.

Reply:
Would it be worthwhile for us to include and source `bash-preexec` if it hasn't already been?

Reply:
Hmmm...now here's an interesting idea.

The big problem I see is that bash-preexec has to be [the absolute last thing sourced in the shell startup](https://github.com/rcaloras/bash-preexec/issues/97), or else it breaks. This would, in turn, require that we have `starship init` be the last line in the user's shell, and even then, `bash-preexec` will [*still* clobber any existing DEBUG traps](https://github.com/rcaloras/bash-preexec/issues/39).

Fundamentally, I don't think there's a way to avoid clobbering `DEBUG`/`PROMPT_COMMAND` in the bash startup because of the limitations of the shell. We can try to take the edge off of it, but there's simply no way to make it foolproof. In light of that, I think we have two options: we can make a best-effort attempt to compensate ourselves and (possibly noisily) disable modules if that fails, or we can try to provide a framework where the user can provide their own initializations.

Reply:
I expect this to be  a pretty rare occurrence--the two Bash config frameworks I'm able to find are [Oh-My-Bash](https://github.com/ohmybash/oh-my-bash) and [Bash-it](https://github.com/Bash-it/bash-it). Oh-my-bash doesn't hook `DEBUG` at all, while `Bash-it` only hooks it specifically to provide a `preexec`-like function (sadly, it uses different function names for that functionality than `bash-preexec`).

What I'm taking from above is that it's going to be pretty rare for a user to hook the DEBUG trap without being aware that they've done it, so if we warn them about it, they should usually know how to fix it. We could enhance this by adding something in our documentation if needed.

Reply:
At this point, I'm thinking we should probably remove Bash support for this feature and circle back to it once we have a robust solution figured out. The rest of the feature seems to work well. 🙂

Reply:
Seems like a plan.

I'll put in some code to have starship log a message when calling init bash and then strip out the init code.

Since we're going to be squash-committing, should I reserve the code somewhere in a comment? Or are we okay with losing the bash init code for now?

Reply:
Could you open an issue to add command duration support in Bash? We could keep the code for it there. 
I'm not at my computer right now, otherwise I'd take care of it.